src/client.ts

```diff
@@ -463,6 +475,225 @@ export class McpdClient {
+   * status before fetching resources. Unhealthy servers are silently skipped to ensure
+   * the method returns quickly without waiting for timeouts on failed servers.
+   *
+   * Servers that don't implement resources (return 501 Not Implemented) are silently
+   * skipped, allowing this method to work with mixed server types.
+   *
+   * Resource fetches from multiple servers are executed concurrently for optimal performance.
+   *
+   * @param options - Optional configuration
+   * @param options.servers - Array of server names to include. If not specified, includes all servers.
+   * @returns Array of resource schemas with transformed names (serverName__resourceName). Only includes resources from healthy servers that support them.
+   * @throws {ConnectionError} If unable to connect to the mcpd daemon
+   * @throws {TimeoutError} If requests to the daemon time out
+   * @throws {AuthenticationError} If API key authentication fails
+   * @throws {McpdError} If health check or initial server listing fails
+   *
+   * @example
+   * ```typescript
+   * // Get all resources from all servers
+   * const allResources = await client.getResources();
+   * // Returns: [
+   * //   { name: "github__readme", uri: "file:///repo/README.md", _serverName: "github", ... },
+   * //   { name: "slack__channels", uri: "slack://...", _serverName: "slack", ... }
+   * // ]
+   *
+   * // Get resources from specific servers only
+   * const someResources = await client.getResources({ servers: ['github'] });
+   *
+   * // Original resource name "readme" becomes "github__readme"
+   * // This prevents clashes if multiple servers have resources with the same name
+   * ```
+   */
+  async getResources(options?: { servers?: string[] }): Promise<Resource[]> {
+    const { servers } = options || {};
+
+    // Get healthy servers (fetches list if not provided, then filters by health).
+    const healthyServers = await this.#getHealthyServers(servers);
+
+    // Fetch resources from all healthy servers in parallel.
+    const results = await Promise.allSettled(
+      healthyServers.map(async (serverName) => ({
+        serverName,
+        resources: await this.#getResourcesByServer(serverName),
+      })),
+    );
+
+    // Process results and transform resource names.
+    const allResources: Resource[] = results.flatMap((result) => {
+      if (result.status === "fulfilled") {
+        const { serverName, resources } = result.value;
+        return resources.map((resource) => {
+          const namespacedResource = {
+            ...resource,
+            name: `${serverName}__${resource.name}`,
+            _serverName: serverName,
+            _resourceName: resource.name,
+            _uri: resource.uri,
+          };
+
+          // Populate resource cache for readResource().
+          this.#resourceCache.set(namespacedResource.name, {
+            serverName,
+            uri: resource.uri,
+          });
+
+          return namespacedResource;
+        });
```

Comment:
nit: updating the cache sounds like a side effect, ideally `.map` should be side effect free, I would add a separate forEach to update the cache.. something like

```ts
const resources = results
  .flatMap(result => transformResultToResource)

resources.forEach(updateCache)

return resources

```